Unify Bubblegum and Taffy moves. Improve flow of Bubblegum section. [[Issue #1045]] (#1060)
hanabi/hanabi.github.io: docs/variant-specific/pink.mdx

<BubblegumFinesse />

### The Bubblegum Prompt

- Normally, you can only _Prompt_ a card if it is the left-most clued card. However, you can _Prompt_ a non-left-most clued card by violating _Pink Promise_. This is called a _Bubblegum Prompt_, which is conceptually similar to a _Bubblegum Finesse_.
- For example, in a 4-player game:
  - All the 1's are played on the stacks.
  - Bob's _Finesse Position_ card has negative pink. He also has three cards clued with number 5 on slot 2, slot 3, and slot 4.
  - Alice clues number 3 to Donald, which focuses a pink 4 on slot 2 and touches a pink 3 on slot 3.
  - Bob sees that in order to make the pink 3 and the pink 4 playable, someone has to have the pink 2. Since no-one else seems to have it, Bob must have it as one of his clued 5's.
  - Given that Alice violated _Pink Promise_, Bob can determine that his pink 2 is on his slot 3. Bob plays his slot 3, and it is the pink 2.
  - Cathy discards.

>   - Donald knows that normally, clues do not cause non-left-most cards to become _Prompted_. Thus, Donald knows that Bob must be responding to Alice's _Bubblegum Prompt_ on a card that is not actually a 3.

Author: actually, I'd probably reword the sentence to something like:

"Donald knows that clues do not normally _Prompt_ a non-leftmost card."

Or drop the double negative, but it loses a bit of it's intention.

>   - The next card after a pink 3 is a pink 4, so Donald knows that the card that Alice focused is a pink 4. Subsequently, since Alice's clue was a _Play Clue_, the pink 4 must be playable. In order for a pink 4 to be playable, someone has to have a pink 3. Donald does not see it anywhere else, so he determines that he must have the pink 3 on slot 3 (since that is the only other clued card that could possibly be a pink 3).

Author: I'm trying to workshop ways to simplify/condense the statements. The first sentence here sounds a bit weird as it's an obvious statement, but it's not clear how it immediately relates to the bubblegum logic.

"Thus, Donald knows that the focus of Alice's _Play Clue_ cannot be a 3; it must be a pink 4. In order for the 4 to be playable, someone must have a pink 3. ..."

- For example, in a 3-player game:
  - It is the first turn and nothing is played on the stacks.
  - Alice clues number 3 to Cathy, touching a pink 2 on slot 5.
  - Bob knows that Alice is violating _Pink Promise_. This must mean that he is supposed to play his slot 3 card.
  - Bob blind-plays slot 3 and it is a red 1.
  - Cathy knows that if this were a _3 Bluff_, then Bob would have played his _Finesse Position_. So it cannot be a _3 Bluff_.
  - Thus, Cathy knows that this must be a _Bubblegum_ clue. Since red 1 is not a pink card and Alice was in _Bluff Seat_ for Bob, this must be a _Bubblegum Bluff_.
  - Cathy marks her slot 5 card as the one-away-from-playable pink card, the pink 2.

<BubblegumBluff />

- _Bubblegum Bluffs_ can only be performed on non-pink cards; otherwise, it will look like a _Bubblegum Finesse_.
- Note that it is impossible to _3 Bluff_ and _Bubblegum Bluff_ at the same time.
- Note that whenever a pink card plays in response to a _Bubblegum_ move, it is guaranteed to the team to be a _Finesse_. (In other words, you cannot use a _Bubblegum Bluff_ with a target of a pink 3 to bluff out a pink 1.)

### The Bubblegum Bluff (Two-Away-From-Playable Form)

> - Normally, _Bubblegum Bluffs_ are always performed by touching _one-way-from-playable_ pink cards.

Author: Sorry, I'm on mobile, so I can't easily find the old language. I really do prefer it though, as I had reviewed that multiple times already before submitting it for review.

Can you think of a less invasive way to update the old language if you feel it needs additional emphasis on the fact that other conventions like 2 saves exist?

Reviewer: not sure, suggest something using the suggest feature

Author: ```
- Note that, as with a standard _Bluff_, the receiver of a _[Bubblegum Bluff](#the-bubblegum-bluff)_ marks their pink card as the lowest-possible, useful value it could be (given the team's preceding actions).
- However, it is sometimes possible that the lowest-possible, useful value is actually _two-away-from-playable_.
```

What about something like this? (I'll find the suggest feature shortly).

- Got rid of "good touch" and replaced it with "useful"
- Added "(given the team's preceding actions)" to hint at the fact that there might be blind plays, etc... a 2 save does not a bluff make :)

Reviewer: > Note that, as with a standard _Bluff_, the receiver of a _[Bubblegum Bluff](#the-bubblegum-bluff)_ marks their pink card as the lowest-possible, useful value it could be (given the team's preceding actions).

That's not how normal bluffs work though. After a bluff, the receiver marks their card as all the matching one-away-from-playable cards + all of the N arbitrary bluffs that were previously agreed upon before the game started (3 bluff, critical 4 bluff, etc.)

Reviewer: Trash bluffs exist, and trash bluffs are done with 1's, so correct me if I am wrong, but the "lowest rank" of the card isn't really relevant, its the N-away-from-playable part.

Author: would it be clearer to hyperlink the text to the bluff docs? https://hanabi.github.io/docs/level-11#the-bluff

I used the word "standard" to not have to worry about all of those other bluffs. is there another word you'd like better?

The "lowest rank" matters when determining the possible values of the target of a bubblegum move. For example, on turn 1, Alice gives Cathy a "2" clue focusing slot 1. Bob blind plays red 1 from slot 2. What is Cathy's note?

"Logically" it could be i3, i4, or i5 (not i1 as Bob would have played from slot 1, and not i2 as Bob would have played from slot 1). We agree that it's the lowest rank of these, in this case i3.

> - Normally, _Bubblegum Bluffs_ are always performed by touching _one-way-from-playable_ pink cards.

Author: ```suggestion
- Note that, as with a standard _Bluff_, the receiver of a _[Bubblegum Bluff](#the-bubblegum-bluff)_ marks their pink card as the lowest-possible, useful value it could be (given the team's preceding actions).
```

Author: Could also drop the leading "Note that, ..." and start with "As with a standard _Bluff_, ...."

> - Normally, _Bubblegum Bluffs_ are always performed by touching _one-way-from-playable_ pink cards.

Author: ```suggestion
- Note that, as with a standard _Bluff_, the receiver of a _[Bubblegum Bluff](#the-bubblegum-bluff)_ marks their pink card as the lowest-possible, useful value it could be (given the team's preceding actions).
```

Author: (oops, dupe suggestion? internet lag I suppose)

> - Rarely, it is possible to perform a _Bubblegum Bluff_ on a _two-away-from-playable_ pink card. However, this can only be done in a specific situation where if the clued card to be _one-way-from-playable_, it would not work out.

Author: ```suggestion
- However, it is sometimes possible that the lowest-possible, useful value is actually _two-away-from-playable_.
```

Author: also wouldn't be opposed to also wording it like:

"However, it sometimes happens that the lowest-possible, ...."

Author: what I want to emphasize is that this not really a "rare" thing. it's just a "numbers + logic" thing depending on what clue is available for what card with what on the stacks.

<BubblegumFinesse />

- Whenever a pink card plays in response to a _Bubblegum_ move, it is guaranteed to the team to be a _Finesse_.

### The Bubblegum Prompt

> - Normally, you can only _Prompt_ a card if it is the left-most clued card. However, you can _Prompt_ a non-left-most clued card by violating _Pink Promise_. This is called a _Bubblegum Prompt_, which conceptually similar to a _Bubblegum Finesse_.

Author: s/which/which is

  - Thus, Cathy knows that this must be a _Bubblegum_ clue. Since red 1 is not a pink card and Alice was in _Bluff Seat_ for Bob, this must be a _Bubblegum Bluff_.
  - Cathy marks her slot 5 card as the one-away-from-playable pink card, the pink 2.

<BubblegumBluff />

- _Bubblegum Bluffs_ can only be performed on non-pink cards; otherwise, it will look like a _Bubblegum Finesse_.
- Note that it is impossible to _3 Bluff_ and _Bubblegum Bluff_ at the same time.

### The Bubblegum Bluff (Two-Away-From-Playable Form)

- Normally, _Bubblegum Bluffs_ are always performed by touching _one-way-from-playable_ pink cards.
- Rarely, it is possible to perform a _Bubblegum Bluff_ on a _two-away-from-playable_ pink card. However, this can only be done in a specific situation where, due to the circumstances of the clue, it would be impossible for the clued card to be _one-way-from-playable_.
- For example, in a 3-player game:
  - It is the first turn and nothing is played on the stacks.
  - Alice clues number 2 to Cathy, touching a pink 3 on slot 3 as a _Play Clue_. (The pink 3 is currently _two-away-from-playable_.)
  - Bob knows that Alice is violating _Pink Promise_.

>   - If Bob blind-plays his _Finesse Position_ card and it successfully plays, it is guaraunteed to be one of the 1s. If that happened, then Cathy would mark her clued card as the matching 2 and will go on to misplay it.

Author: (oops, I need to fix this typo... will do that sometime before submission.)

Reviewer: what is the typo?

Author: some check was failing: "guaranteed" only has one "u"

<BubblegumFinesse />

> - _Bubblegum Finesses_ can only be used to get pink cards.

Reviewer: why is this line deleted

Author: it's replaced with the line below it. I thought they were identical, with the second one being clearer to someone who is learning. 

given your comment on that line as well, it seems that may not be the case?

how do you interpret this (deleted) line?
"Bubblegum Finesses can only be used to get pink cards."

what is it actually saying?
I think the only thing it _can_ be saying is that once a pink card plays, it's a Finesse.

Author: hmm... did we accidentally remove the other statement?

I think it's important to communicate somewhere that Bubblegum moves causing pink cards to blind play are finesses.  That might now be missing from the doc? or is it elsewhere?

Author: (oh, we do have "- _Bubblegum Bluffs_ can only be performed on non-pink cards; otherwise, it will look like a _Bubblegum Finesse_." under Bubblegum Bluffs... but hmmm, is that clear enough?)

Reviewer: i suspect so

> - Whenever a pink card plays in response to a _Bubblegum_ move, it is guaranteed to the team to be a _Finesse_.

Reviewer: i dont understand this line.
isn't it false because bubblegum bluffs exists?

Author: as far as I understood, there is no such thing as bubblegum-bluffing a pink (or omni) card. those are finesses. (hence the reword above.)

similar to Bob playing red 1 into a red clue on Cathy's red 3. it's not a bluff. either Cathy bombs red 3 or Bob bombs slot 2 next round.

- _Bubblegum Bluffs_ can only be performed on non-pink cards; otherwise, it will look like a _Bubblegum Finesse_.
- Note that it is impossible to _3 Bluff_ and _Bubblegum Bluff_ at the same time.

### The Bubblegum Bluff (Two-Away-From-Playable Form)

- Normally, _Bubblegum Bluffs_ are always performed by touching _one-way-from-playable_ pink cards.

> - Rarely, it is possible to perform a _Bubblegum Bluff_ on a _two-away-from-playable_ pink card. However, this can only be done in a specific situation where, due to the circumstances of the clue, it would be impossible for the clued card to be _one-way-from-playable_.

Author: @Zamiell obviously, you thought my proposed wording was unclear, hence the changes 😛 

but I sort of disagree with this wording (which reminds me more of the pre-PR document).
it's not really a "specific situation" or "circumstance" -- it's precisely the _logic_ following from _Good Touch_ and lowest-value.

what were your thoughts on this?

Reviewer: does it have to do with "good touch" though?
like, say for example you do a "2 save" and it causes a crazy blind play, then it can't be a 2 save anymore, and it is a taffy bluff as a 3.
in that example, the operant principle is the fact that the "2 save" convention already exists and that since it exists, it wouldn't cause any blind plays.
but the doc can't be worded like this:
> Sometimes, a clue would look like a _2 Save_, but it can't be a _2 Save_, so then it would have to be on the next _two-away-from-playable_ card.
so we need to word that sentence in a more generic way, so i chose the phase "circumstances of the clue", but you can workshop this if you don't like it.

Author: ah okay I see what you're saying.

the "logic" relies on other conventions as well, like 2 saves existing.

but in a way, that's the point. this convention is built on top of those existing ones. we know that a 2 save doesn't cause blind plays. so if there is a blind play, the card can't be a 2.

and we agree that lying about a card's identity needs to be fixed in a certain way (blind play from number slot).

given those facts, the card being one away or two away isn't really important -- it's the lowest possible value it could be, assuming the card is useful ("good touch").

That's precisely where (lowest possible and) good touch comes in: "what value is the card?"

Reviewer: I updated the doc a bit if you want to review the pr again and suggest more edits (maybe we should use the GitHub suggest edit feature, if you're familiar with that)

Author: sorry been traveling, on mobile right now, I can take a look and try to suggest something

  - Bob blind-plays his slot 2, but it is a red 1. It is now revealed to Bob that this was a _Bubblegum Bluff_ instead of a _Bubblegum Finesse_.
  - Cathy knows that if her clued card were really a 2, Bob would have played slot 1. Thus, it cannot be a 2. Furthermore, there are no _Ejections_ that match this situation, so a _Bubblegum_ move must have happened. Normally, a _Bubblegum Bluff_ would be on a _one-away-from-playable_ card, but that would be the pink 2, and due to the circumstances of the clue, it is impossible to be a 2. Thus, it instead must be the _two-away-from-playable_ card, which is the pink 3.

### The Bubblegum Double Bluff

- Normally, after a _Bubblegum Bluff_, the entire team is synchronized on the identity of the focus of the _Bubblegum_ move.
- But what if the note that the clue receiver would write on their card is still incorrect? In this situation, the clue-giver expects additional blind-plays to resynchronize the information, in the same way that a normal _Double Bluff_ works.
- For example, in a 4-player game:
  - It is the first turn and nothing is played on the stacks.
  - Alice clues number 3 to Donald, touching a pink 4 on slot 1.

>   - Bob knows that Alice is violating _Pink Promise_. This must mean that he is supposed to play his slot 3 card as pink 1, with pink 2 and 3 promised as playable through his current _Finesse Position_ and _Second Finesse Position_, respectively.

Author: (d'oh, I just realized you have made edits to the PR...)

I think my original intention was to say that this is a finesse, hence the word "through". it could be layered, etc. 

not sure if the new language about "Second Finesse Position" and "respectively" is allowing for that? it's probably "clear enough" ? but maybe worth being extra clear by not mentioning it.

Reviewer: > it could be layered

yeah but all finesses can be layered, so I don't think we need to specifically mention that.

Author: My point was that the language "through his _Finesse Position_" should be sufficient, without mentioning second finesse, since that may not even be where the card is located. I.e. the current language is more specific and hence inaccurate in some cases.

### The Bubblegum Bluff (Two-Away-From-Playable Form)

- Normally, _Bubblegum Bluffs_ are always performed by touching _one-way-from-playable_ pink cards.

> - However, it is sometimes possible that the lowest-possible, useful value is actually _two-away-from-playable_.

Author: just a heads up that the language "lowest-possible, useful value" here might be incomplete/lack context if we aren't also taking the rewrite suggestion for the first line.

Author: "...lowest-possible, useful value of a focused pink card is actually..." ?
